modules/javafx.graphics/src/jslc/java/com/sun/scenario/effect/compiler/backend/hw/SLBackend.java

```diff
@@ -81,8 +81,20 @@ protected final void output(String s) {
         sb.append(s);
     }
 
+
+    static String PIXCOORD = "vec2 pixcoord = vec2(\n"+
+        "    gl_FragCoord.x-jsl_pixCoordOffset.x,\n" +
+        "    ((jsl_pixCoordOffset.z-gl_FragCoord.y)*jsl_pixCoordOffset.w)-jsl_pixCoordOffset.y);\n";
```

Comment:
This GLSL shader code doesn't belong here. `SLBackend` is the super-class for both of the `HLSLBackend` (DirectX) and `GLSLBackend` (OpenGL). If there is no way to currently do what is needed entirely in `GLSLBackend`, then one possible approach might be to make `SLBackend::getShader` non-final, and override it in `GLSLBackend`

Reply:
Sounds like the latter would be the preferred choice as the bugs are reported for opengl on armhf and x86_64 specifically, and Its unclear if this could be an issue for directx or not.

Reply:
I'll refactor in the morning, and update unless @johanvos has any further thoughts on this?

Reply:
iirc, the bug is specific to some drivers that are used on linux-based systems, so I don't think it will show in platforms using DirectX. It's a pity we need separate code for the HLSLBackend and the GLSLBackend but it's probably the cleanest way to address a driver issue like this.